src/element-service.js

```diff
@@ -85,3 +91,60 @@ function isElementScheduled(win, elementName) {
   }
   return !!win.ampExtendedElements[elementName];
 }
+
+
+/**
+ * Returns a promise for a service for the given id and window. Also expects
+ * an element that has the actual implementation. The promise resolves when
+ * the implementation loaded.
+ * Users should typically wrap this as a special purpose function (e.g.
+ * viewportForDoc(...)) for type safety and because the factory should not be
+ * passed around.
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
+ * @param {string} id of the service.
+ * @param {string} providedByElement Name of the custom element that provides
+ *     the implementation of this service.
+ * @return {!Promise<*>}
+ */
+export function getElementServiceForDoc(nodeOrDoc, id, providedByElement) {
+  return getElementServiceIfAvailableForDoc(nodeOrDoc, id, providedByElement)
+      .then(service => {
+        return user().assert(service,
+              'Service %s was requested to be provided through %s, ' +
+              'but %s is not loaded in the current page. To fix this ' +
+              'problem load the JavaScript file for %s in this page.',
+              id, providedByElement, providedByElement, providedByElement);
+      });
+}
+
+/**
+ * Same as getElementService but produces null if the given element is not
+ * actually available on the current page.
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
+ * @param {string} id of the service.
+ * @param {string} providedByElement Name of the custom element that provides
+ *     the implementation of this service.
+ * @return {!Promise<?Object>}
+ */
+export function getElementServiceIfAvailableForDoc(
```

Comment:
There wasn't any tests for this file at all, should I create one or file an issue to follow up on it?